apartments/tests.py

```diff
@@ -1,85 +1,47 @@
-        city=new_city,
-        address='street',
-        rent=4500,
-        num_of_roomates=2,
-        num_of_rooms=3,
-        start_date='2021-1-1',
-        about='Hey!',
-        image_url='www.some-url.com',
-    )
-    apartment.save()
-    return apartment
-
-
 @pytest.mark.django_db
 class TestModels:
 
-    def test_city_creation(self, new_city):
-        assert new_city is not None
-        assert new_city.cityName == 'some_city'
+    def test_city_creation(self, city_model):
+        assert city_model is not None
+        assert city_model.cityName == 'nice_city'
 
-    def test_get_city(self, new_city):
-        check_city = City.objects.get(cityName=new_city.cityName)
-        assert check_city == new_city
+    def test_get_city(self, city_model):
+        check_city = City.objects.get(cityName=city_model.cityName)
+        assert check_city == city_model
 
-    def test_apartment_creation(self, new_apartment):
-        assert new_apartment is not None
-        assert new_apartment.address == 'street'
-        assert new_apartment.rent == 4500
-        assert new_apartment.is_relevant is True
+    def test_apartment_creation(self, apartment_model):
+        assert apartment_model is not None
+        assert apartment_model.address == 'street'
+        assert apartment_model.rent == 4500
+        assert apartment_model.is_relevant is True
 
-    def test_get_apartment(self, new_apartment):
-        check_apartment = Apartment.objects.get(owner=new_apartment.owner)
-        assert check_apartment == new_apartment
+    def test_get_apartment(self, apartment_model):
+        check_apartment = Apartment.objects.get(owner=apartment_model.owner)
+        assert check_apartment == apartment_model
 
-    def test_get_apartment_by_id(self, new_apartment):
-        check_apartment_success = Apartment.get_apartment_by_id(new_apartment.owner.id)
+    def test_get_apartment_by_id(self, apartment_model):
+        check_apartment_success = Apartment.get_apartment_by_id(apartment_model.owner.id)
         check_apartment_fail = Apartment.get_apartment_by_id(-1)
-        assert check_apartment_success == new_apartment
+        assert check_apartment_success == apartment_model
         assert check_apartment_fail is None
 
 
 @pytest.mark.django_db
 class TestViews:
 
-    def test_update_apartment_view_to_owner(self, client, new_apartment):
-        client.login(email='test@test.com', password='password')
+    def test_update_apartment_view_to_owner(self, client, apartment_model):
+        client.login(email='apartmentemail@address.com', password='password')
         response = client.get('/apartments/update')
         assert response.status_code == 200
 
-    def test_update_apartment_view_to_not_owner_user(self, client, new_user):
-        client.login(email='test@test.com', password='password')
+    def test_update_apartment_view_to_not_owner_user(self, client, user_model):
+        client.login(email='email@address.com', password='password')
```

Comment:
You are passing in user_model as a parameter and it isn't used.

Reply:
same as above, ill try running it without it

Reply:
Does not work without the fixture